components/FeedList/FeedList.jsx

```diff
@@ -1,57 +1,85 @@
-import { useRoute } from '@react-navigation/native';
+import React, {useState, useCallback, useEffect, useContext} from 'react';
 import PropTypes from 'prop-types';
-import { FlatList, View } from 'react-native';
-import { verticalScale } from '../../assets/metrics/Metrics';
+import {useRoute} from '@react-navigation/native';
+import {FlatList, View} from 'react-native';
 import Photos from '../../screens/ProfileScreen/page/Photos/Photos';
 import SkeletonPhotos from '../Skeleton/SkeletonPhotos';
 import SkeletonPosts from '../Skeleton/SkeletonPosts';
 import Warning from '../Warning/Warning';
 import FeedItem from './FeedItem';
+import Search from '../Search/Search';
+import {GlobalDataContext} from '../../contexts/context';
 import styles from './style';
 
-const FeedList = ({ state, navigation, loading, showHide, screen }) => {
+const FeedList = ({navigation, showHide}) => {
   const route = useRoute();
+  const {feeds} = useContext(GlobalDataContext);
+  const [data, setData] = useState(feeds);
+  const [emptyDataMessage, setEmptyDataMessage] = useState('');
+  const [isLoaded, setIsLoaded] = useState(true);
+
+  const loadedData = () => {
+    const timer = setTimeout(() => setIsLoaded(false), 2500);
+    return () => clearTimeout(timer);
+  };
+
+  useEffect(() => {
+    loadedData();
+  }, []);
 
   const renderSwitchValue = (item, index) => {
-    if (route.name === 'Profile') {
-      if (showHide) {
-        return loading ? (
+    switch (route.name) {
+      case "Profile":
+        if (showHide) {
+          return isLoaded ? (
+            <SkeletonPosts />
+          ) : (
+            <FeedItem item={item} itemIndex={index} navigation={navigation} />
+          );
+        } else {
+          return isLoaded ? <SkeletonPhotos /> : <Photos item={item} />;
```

Comment:
?

Reply:
ignore this

Reply:
you don't need switch if route.name only has one value: Profile

Reply:
ok, I removed the switch and wrote it differently, the given component is drawn on two pages and for that I wrote this function as a condition.